api/src/pcapi/routes/adage_iframe/favorites.py

```diff
@@ -0,0 +1,59 @@
+from pcapi.core.educational import exceptions as educational_exceptions
+from pcapi.core.educational import repository as educational_repository
+from pcapi.core.educational.api import favorites as educational_api
+from pcapi.core.educational.repository import find_redactor_by_email
+from pcapi.models.api_errors import ApiErrors
+from pcapi.routes.adage_iframe import blueprint
+from pcapi.routes.adage_iframe.security import adage_jwt_required
+from pcapi.routes.adage_iframe.serialization import favorites as serializers
+from pcapi.routes.adage_iframe.serialization.adage_authentication import AuthenticatedInformation
+from pcapi.routes.adage_iframe.serialization.favorites import CollectiveOfferFavoritesBodyModel
+from pcapi.serialization.decorator import spectree_serialize
+
+
+@blueprint.adage_iframe.route("/collective/offer-favorites", methods=["POST"])
+@spectree_serialize(response_model=serializers.FavoritesOfferResponseModel, api=blueprint.api)
+@adage_jwt_required
+def post_collective_offer_favorites(
+    authenticated_information: AuthenticatedInformation,
+    body: CollectiveOfferFavoritesBodyModel,
+) -> serializers.FavoritesOfferResponseModel:
+    try:
+        offer = educational_repository.get_collective_offer_by_id(offer_id=body.offerId)
+    except educational_exceptions.CollectiveOfferNotFound:
+        raise ApiErrors({"offer": "l'offre est introuvable"}, status_code=404)
+
+    redactor = find_redactor_by_email(authenticated_information.email)  # type: ignore [arg-type]
```

Comment:
Un peu dommage ces deux type ignore, est-ce qu’on pourrait pas juste mettre un `assert authenticated_information.email is not None` pour donner un coup de main à mypy si on est certain qu’à ce point ce field ne peut être null, ou si ça peut arriver peut être juste raise la même ApiError que ci-dessous puisque dans tous les cas si email vaut None `find_redactor_by_email` renverra None